From 0acf3076632d4808b92716031239824a43926b04 Mon Sep 17 00:00:00 2001
From: Stanislaw Gruszka <sgruszka@redhat.com>
Date: Wed, 10 Dec 2014 10:29:20 -0500
Subject: [wireless] iwlwifi/mvm: set MAC_FILTER_IN_BEACON correctly for STA/P2P client

Message-id: <1418207389-7899-7-git-send-email-sgruszka@redhat.com>
Patchwork-id: 101578
O-Subject: [RHEL7.1 PATCH 06/35] iwlwifi: mvm: set MAC_FILTER_IN_BEACON correctly for STA/P2P client
Bugzilla: 1155538
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: John Greene <jogreene@redhat.com>

From: Luciano Coelho <luciano.coelho@intel.com>

BZ#1155538

commit 7c8b3bc688bdcfd7789efbd17da4d85b104ad351 upstream.

In commit cad3f08c (iwlwifi: mvm: enable MAC_FILTER_IN_BEACON when
forced_assoc_off is set) the code to set the MAC_FILTER_IN_BEACON flag
was accidentally moved to the main block of the if statement, while it
should be in the else block instead.  Move it to the right place.

Fixes: cad3f08c23de ("iwlwifi: mvm: enable MAC_FILTER_IN_BEACON when forced_assoc_off is set")
Reviewed-by: Johannes Berg <johannes.berg@intel.com>
Signed-off-by: Luciano Coelho <luciano.coelho@intel.com>
Signed-off-by: Emmanuel Grumbach <emmanuel.grumbach@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Brew:
https://brewweb.devel.redhat.com/taskinfo?taskID=8330268

Testing:
Tested on iwl7265 adapter.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/wireless/iwlwifi/mvm/mac-ctxt.c b/drivers/net/wireless/iwlwifi/mvm/mac-ctxt.c
index 8b79081..db84533 100644
--- a/drivers/net/wireless/iwlwifi/mvm/mac-ctxt.c
+++ b/drivers/net/wireless/iwlwifi/mvm/mac-ctxt.c
@@ -720,11 +720,6 @@ static int iwl_mvm_mac_ctxt_cmd_sta(struct iwl_mvm *mvm,
      !force_assoc_off) {
   u32 dtim_offs;
 
-  /* Allow beacons to pass through as long as we are not
-   * associated, or we do not have dtim period information.
-   */
-  cmd.filter_flags |= cpu_to_le32(MAC_FILTER_IN_BEACON);
-
   /*
    * The DTIM count counts down, so when it is N that means N
    * more beacon intervals happen until the DTIM TBTT. Therefore
@@ -758,6 +753,11 @@ static int iwl_mvm_mac_ctxt_cmd_sta(struct iwl_mvm *mvm,
   ctxt_sta->is_assoc = cpu_to_le32(1);
  } else {
   ctxt_sta->is_assoc = cpu_to_le32(0);
+
+  /* Allow beacons to pass through as long as we are not
+   * associated, or we do not have dtim period information.
+   */
+  cmd.filter_flags |= cpu_to_le32(MAC_FILTER_IN_BEACON);
  }
 
  ctxt_sta->bi = cpu_to_le32(vif->bss_conf.beacon_int);
-- 
1.7.1